cpp/include/cudf/detail/sizes_to_offsets_iterator.cuh

```diff
@@ -317,7 +318,8 @@ std::pair<std::unique_ptr<column>, size_type> make_offsets_child_column(
   auto const total_elements = sizes_to_offsets(input_itr, input_itr + count + 1, d_offsets, stream);
   CUDF_EXPECTS(
     total_elements <= static_cast<decltype(total_elements)>(std::numeric_limits<size_type>::max()),
```

Comment:
I was worried that the other uses of this pattern always explicitly cast to `int64_t` event though the output of `sizes_to_offsets` can be either `uint64_t` or `int64_t`. However, after reading all the implicit promotion and binop conversion rules a number of times, and looking at cppinsights, I think both patterns are safe (since in the case that one operand is unsigned the signed operand will be promoted which is not lossy).

In fact, I _think_ that is is safe to just have [`total_elements <= std::numeric_limits<size_type>::max()`](https://cppinsights.io/lnk?code=I2luY2x1ZGUgPGxpbWl0cz4KI2luY2x1ZGUgPGNzdGRpbnQ+CgppbnQgbWFpbih2b2lkKQp7CiAgICB1aW50NjRfdCB1YSA9IDEwOwogICAgaW50NjRfdCBpYSA9IDEwOwogICAgYXV0byBpYiA9IHN0ZDo6bnVtZXJpY19saW1pdHM8aW50Pjo6bWF4KCk7CiAgICBhdXRvIHViID0gc3RkOjpudW1lcmljX2xpbWl0czxpbnQ+OjptYXgoKTsKICAgIGF1dG8gdXUgPSB1YSA8PSB1YjsKICAgIGF1dG8gdWkgPSB1YSA8PSBpYjsKICAgIGF1dG8gaXUgPSBpYSA8PSB1YjsKICAgIGF1dG8gaWkgPSBpYSA8PSBpYjsKICAgIHJldHVybiAwOwp9&insightsOptions=cpp17,all-implicit-casts&std=cpp17&rev=1.0) although it's probably nicer to keep the casts to help readers.